documentation/api/query/examples-pql.markdown

```diff
@@ -0,0 +1,447 @@
+```
+
+*Output:*
+
+``` json
+[
+    {
+        "certname": "greenserver.vm"
+    }
+]
+```
+
+***
+
+### Fact and resource filtering for classes
+
+CentOS boxes with the `apache` Puppet class.
+
+``` ruby
+inventory[certname] { facts.operatingsystem = "CentOS" and
+                      resources { type = "Class" and
+                                  title = "apache" } }
+```
+
+*Output:*
+
+``` json
+[
+    {
+        "certname": "greenserver.vm"
+    }
+]
+```
+
+***
+
+### Fact, resource and environment filtering
+
+All windows servers, with service `sqlserver` in a particular feature branch environment.
+
+``` ruby
+nodes { certname in inventory[certname] { facts.osfamily = "Windows" } and
+        resources { type = "Service" and
+                    title = "sqlserver" } and
+        report_environment = "feature_SYS-4926" }
+```
+
+*Output:*
+
+``` json
+[ {
+  "deactivated" : null,
+  "latest_report_hash" : "4a956674b016d95a7b77c99513ba26e4a744f8d1",
+  "facts_environment" : "laboratory",
+  "cached_catalog_status" : "not_used",
+  "report_environment" : "laboratory",
+  "latest_report_corrective_change" : false,
+  "catalog_environment" : "laboratory",
+  "facts_timestamp" : "2016-07-18T04:12:12.912Z",
+  "latest_report_noop" : false,
+  "expired" : null,
+  "latest_report_noop_pending" : null,
+  "report_timestamp" : "2016-07-18T04:12:15.907Z",
+  "certname" : "windowserver.vm",
+  "catalog_timestamp" : "2016-07-18T04:12:12.917Z",
+  "latest_report_status" : "success"
+} ]
+```
+
+***
+
+### Timestamp filtering
+
+Nodes that haven't checked in for 7 days.
+
+``` ruby
+nodes { report_timestamp <= "2016-08-03 00:00:00" }
+```
+
+*Output:*
+
+``` json
+[
+    {
+        "cached_catalog_status": "not_used",
+        "catalog_environment": "production",
+        "catalog_timestamp": "2016-08-15T11:06:26.275Z",
+        "certname": "greenserver.vm",
+        "deactivated": null,
+        "expired": null,
+        "facts_environment": "production",
+        "facts_timestamp": "2016-08-15T11:06:26.140Z",
+        "latest_report_corrective_change": null,
+        "latest_report_hash": "4a956674b016d95a7b77c99513ba26e4a744f8d1",
+        "latest_report_noop": false,
+        "latest_report_noop_pending": null,
+        "latest_report_status": "changed",
+        "report_environment": "production",
+        "report_timestamp": "2016-08-15T11:06:18.393Z"
+    }
+]
+```
+
+***
+
+### Mcollective sub-collective filtering
+
+Show nodes that belong in the sub-collective `dc1`.
+
+``` ruby
+inventory[certname] { facts.mcollective.server.collectives[*] = "dc1" } }
+```
+
+*Output:*
+
+``` json
+[
+    {
+        "certname": "greenserver.vm"
+    },
+    {
+        "certname": "yellowserver.vm"
+    }
+]
+```
+
+***
+
+### Profile querying
+
+Show nodes that have the profile class `Profile::Remote_mgmt` applied to it.
+
+``` ruby
+nodes { resources { type = "Class" and title = "Profile::Remote_mgmt" } }
+```
+
+*Output:*
+
+``` json
```

Comment:
Umm, probably, but @nfagerlund only let me know about it within the last few months I think. Nice huh? We need a pql variant, I'm cheating and using ruby because its close enough.